### PR TITLE
Validate Thingy:91 with all samples

### PR DESCRIPTION
**Pull request by szczys**
This pull request collects the changes needed to make the Thingy:91 work with all Golioth samples.

<a href="https://gitpod.io/#https://github.com/golioth/zephyr-sdk/pull/165"></a>

### CODE REVIEW

File: samples/dfu/README.rst
Comment (mniestroj) [inline samples/dfu/README.rst:247]: Those lines should be fixed in separate commit. Bugfixes separately, new features separately.

File: samples/dfu/README.rst
Comment (mniestroj) [inline samples/dfu/README.rst:23]: We do actually support nRF9160 DK as well, so maybe it is good to add it here as well (in separate commit of course).

File: samples/dfu/README.rst
Comment (mniestroj) [inline samples/dfu/README.rst:12]:
```suggestion
built with a new version number, registered as an artifact, and deployed as a
```

File: samples/dfu/README.rst
Comment (mniestroj) [inline samples/dfu/README.rst:15]: This is an improvement of documentation and is unrelated to "feat: add support for Thingy:91 to dfu sample" commit title. One logical change == one commit. This commit should result at least with following commits:
- `samples: dfu: fix dfu artifact and release commands`
- some other documantation fixes
- `samples: dfu: improve high-level sample description`
- `samples: dfu: add support for Thingy:91 board`

Please remember to first push fixes, then improvements. Preferably bugfixes and/or documentation improvements should be in a separate PR, because they are not related directly to inclusion of Thingy:91 board and having everything in single PR will just block bugfixes from being included.

File: samples/dfu/README.rst
Comment (mniestroj) [inline samples/dfu/README.rst:26]: What do you want to reference here? IMO Golioth docs should refer to SDK, not the other way.

File: samples/dfu/README.rst
Comment (mniestroj) [inline samples/dfu/README.rst:40]: This sounds like a generic doc improvement, so should be in separate commit. Hard to say if it is valid in current form, as there are too many changes as part of single commit.

File: samples/dfu/README.rst
Comment (mniestroj) [inline samples/dfu/README.rst:49]: We are literally entering "bootloader", not some "bootloader mode" in application. I have mixed feeling whether adding "mode" here is and improvement or not. Just a doubt from my side, I won't block this though.

File: samples/dfu/README.rst
Comment (mniestroj) [inline samples/dfu/README.rst:47]: Why don't we want to reference official documentation? Would be great to write down that reason in commit description.

File: samples/dfu/README.rst
Comment (mniestroj) [inline samples/dfu/README.rst:153]: Why adding `-p` everywhere? I don't think that Zephyr docs include that parameter, so why should we be different? If you will convince Zephyr maintainers to include that parameter in every command in docs, then I won't object to include that here. Let's be consistent.

File: samples/logging/README.rst
Comment (mniestroj) [inline samples/logging/README.rst:161]: There should be a reason to use `newtmgr` instead of `mcumgr`. Please put this change in separate commit and make sure to write down the reason of this change. Otherwise we will play ping-pong with switching between `mcumgr` and `newtmgr` if there is no real reason of using one or the other.

File: samples/logging/README.rst
Comment (mniestroj) [inline samples/logging/README.rst:145]: It seems inconsistent to have that sentence only under `nRF9160 Feather or Thingy:91` subsection. In ALL cases we need to open a terminal change working directory.

> locate the directory that contains the sample folder (i.e., ``~/zephyr-nrf/modules/lib/golioth``)

This is the root directory of SDK repo. Sample folder is under `samples/<whatever>/`, so this is misleading now.

File: samples/lightdb_led/README.rst
Comment (mniestroj) [inline samples/lightdb_led/README.rst:222]: I think that there should be a one way direction of dependencies, not back and forth. In this specific case I think that `https://docs.golioth.io` should rather reference what is documented in SDK repo, not the other way.

File: samples/lightdb_led/README.rst
Comment (mniestroj) [inline samples/lightdb_led/README.rst:183]: Good catch. Deserves separate commit though. It is also a good idea to reference commit ID of the change that removed counter usage in code, so that it is easy to track down when documentation went out of sync. This is quite useful for maintenance of stable releases (which we don't do yet, but there are plans to start doing them).

**Review by mniestroj — CHANGES_REQUESTED** [review]
Hard to review due to mixed bugfixes, improvements and new features, all as part of single commit.

Please consider to create a separate PR with bugfixes (and optionally obvious docs improvements). Then create another PR on top with the features you want to add (inclusion of Thingy:91).

Having `feat: ` in commit titles doesn't seem to have any real meaning, as those commits are including bugfixes anyway. This means that the title is a "lie". Anyway, we reference Zephyr commit/contribution guidelines (https://docs.zephyrproject.org/latest/contribute/index.html#commit-guidelines) and actually the first point says:
> A short and descriptive subject line that is less than 72 characters, followed by a blank line. The subject line must include a prefix that identifies the subsystem being changed, followed by a colon, and a short title, for example: doc: update wiki references to new site. (If you’re updating an existing file, you can use git log <filename> to see what developers used as the prefix for previous patches of this file.)

so `feat: ` is obviously not a good idea to be used in commit title.

File: samples/logging/README.rst
Comment (szczys) [reply samples/logging/README.rst:161]: Okay, will do. When flashing the feather, the version of newtmgr that we recommend in our docs flashes much faster. We've normalized docs to use newtmgr for nRF91 examples.

File: samples/logging/README.rst
Comment (szczys) [reply samples/logging/README.rst:145]: Good point, this would make more sense higher in the file to apply to all hardware compiling examples.

I made this change because our build instructions pass the `samples/<whatever`. If user is already in the directory for that sample, build will fail:
```
ERROR: source directory samples/lightdb/get does not exist
FATAL ERROR: refusing to proceed without --force due to above error
```
I think it makes more sense to new users to build from the same root directory every time.

File: samples/dfu/README.rst
Comment (szczys) [reply samples/dfu/README.rst:23]: This opens a larger question: do we add build command examples for every supported hardware?

File: samples/logging/README.rst
Comment (mniestroj) [reply samples/logging/README.rst:145]: > I think it makes more sense to new users to build from the same root directory every time.

Makes sense!

File: samples/dfu/README.rst
Comment (mniestroj) [reply samples/dfu/README.rst:26]: > I think that information is too much to be included in the readme, 

I agree.

> but since we have documentation on it, we should make that known.

There are plenty of sites that already document that. I would rather stick closer to the source/vendor, i.e. reference Nordic (author of nrfjprog) and Jared (modified newtmgr with downstream patches) documentation instead.

Do we really want to break "Golioth docs ------> Golioth Zephyr SDK ------> Zephyr RTOS" one way relation? Isn't that going to make us pain later when synchronizing one to the other?

File: samples/dfu/README.rst
Comment (mniestroj) [reply samples/dfu/README.rst:23]: Good question. In general `samples/` in this repo should be just samples of what we implement in this repo, which is sample usage of lightdb, stream, dfu, etc. In the beginning we just had 3 platforms that we supported, so it made sense to give overlay files for each platform. Currently this list is much longer and the maintenance effort grows with each added platform, as in theory we should make sure that each commit still works on each platform.

File: samples/logging/README.rst
Comment (szczys) [reply samples/logging/README.rst:161]: I started changing over the docs to use `mcumgr` instead of `newtmgr`. But then in testing I found that when an upload stalls (fairly frequently), you cannot stop the `mcumgr` process with CTRL-C; it must be killed from another terminal ([there's an open issue](https://github.com/apache/mynewt-mcumgr-cli/issues/21)).

This is a pretty bad user experience. On the one hand, `mcumgr` is an easier install process, but on the other hand `newtmgr` can be stopped with a keyboard interrupt.

Any thoughts on this?

**Comment by szczys** [comment]
I've restructured this PR to isolate the changes related to adding Thingy:91 support.

This PR is rebased on #169 which should be merged first.

**Comment by mniestroj** [comment]
@szczys Can you rebase this PR, so that merge conflicts are resolved?

One of the things I am missing now is build tests for this platform. We do have build tests for all other platforms, so that we don't accidentaly break them.

Additionally, we there is a [Device Matrix](https://github.com/golioth/zephyr-sdk#device-matrix) to be updated in main README file.

Last, but most important, are you commited to testing this platform regularly and maintaining support for it? IMO we already have too many platforms with no automated runtime tests, so I don't want to take that on me :)

**Comment by szczys** [comment]
> Can you rebase this PR, so that merge conflicts are resolved?

Yes, done!

> One of the things I am missing now is build tests for this platform. We do have build tests for all other platforms, so that we don't accidentaly break them.

I haven't looked into these but with some guidance I'm happy to work on this.

> Additionally, we there is a [Device Matrix](https://github.com/golioth/zephyr-sdk#device-matrix) to be updated in main README file.

I can work on that. Is the SDK version the hash for this commit once it's merged? Where do I get golioth version info?

> Last, but most important, are you commited to testing this platform regularly and maintaining support for it? IMO we already have too many platforms with no automated runtime tests, so I don't want to take that on me :)

Sure? I agree that autoated runtime tests are important and we need to solve that. For now, I can run the tests on this device when needed.

**Comment by szczys** [comment]
I made an update to the device matrix and added yaml entries in the samples for thingy91_nrf9160_ns.

**Comment by szczys** [comment]
> However I think that we should assume that users understand west when working with SDK repo directly (as opposed to following docs repo).

It would be better assume the user understands enough to remove the `-p`. This is a possible tripping point when trying out Golioth. I was confused when I first encountered the need for a pristine build, especially since there is no `west clean` command (I'm used to typing `make clean`).

But, we can debate this another time. I'll remove these and push a commit so we can get this PR merged. Thanks!

File: samples/lightdb_led/README.rst
Comment (mniestroj) [inline samples/lightdb_led/README.rst]: Could you please squash `-p` removal? There is no point in introducing it and removing right away in subsequent commit in the same PR.

Comment (szczys) [reply samples/lightdb_led/README.rst]: Done! Please take a look when you have a chance. Thanks!

**Review by mniestroj — APPROVED** [review]

**Comment by mniestroj** [comment]
@szczys Thanks a lot!

**Pull request merged**
